pkg/pagerduty/pagerduty_test.go

```diff
@@ -0,0 +1,174 @@
+package pagerduty
+
+import (
+	"context"
+
+	pdApi "github.com/PagerDuty/go-pagerduty"
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	pdMock "github.com/openshift/backplane-cli/pkg/pagerduty/mocks"
+)
+
+var _ = Describe("Pagerduty", func() {
+	var (
+		mockCtrl        *gomock.Controller
+		mockPdClient    *pdMock.MockPagerDutyClient
+		pagerDuty       *PagerDuty
+		testIncidentID  string
+		testClusterID   string
+		testServiceID   string
+		testClusterName string
+		testAlertName   string
+	)
+
+	BeforeEach(func() {
+		mockCtrl = gomock.NewController(GinkgoT())
+		mockPdClient = pdMock.NewMockPagerDutyClient(mockCtrl)
+		pagerDuty = NewPagerDuty(mockPdClient)
+		testIncidentID = "incident-id-000"
+		testClusterID = "cluster-id-000"
+		testServiceID = "service-id-000"
+		testClusterName = "test-cluster-name"
+		testAlertName = "alert-error-budget"
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+	})
+
+	Context("When pagerduty client executes", func() {
+		It("Should return incidents", func() {
+
+			// Mock alert response
+			alertResponse := &pdApi.ListAlertsResponse{
+				Alerts: []pdApi.IncidentAlert{
+					alert(
+						testIncidentID,
+						testServiceID,
+						testAlertName,
+						testClusterID,
+						"triggered",
+					),
+				},
+			}
+
+			// Mock the service response
+			serviceResponse := &pdApi.Service{
+				Description: testClusterName,
+			}
+
+			mockPdClient.EXPECT().ListIncidentAlerts(testIncidentID).Return(alertResponse, nil).Times(1)
+			mockPdClient.EXPECT().GetServiceWithContext(context.TODO(), testServiceID, gomock.Any()).Return(serviceResponse, nil).Times(1)
+
+			alerts, err := pagerDuty.GetIncidentAlerts(testIncidentID)
+			Expect(err).To(BeNil())
+			Expect(len(alerts)).To(Equal(1))
+
```

Comment:
A small suggestion, shall we also add a small check using `formatAlert` function ? It's used inside a for loop in GetIncidentAlerts which is appending formatted alerts to the main alerts struct.
We can also add a check for `RateLimited`.

Reply:
Added separate test case to cover format func